### DISCUSSION TITLE
Inquiry about Prototyping Deck and Servo Motor Attachment for Crazyflie 2.0

### DISCUSSION THREAD

**Discussion by NarekHarutyu** [bitcraze/discussions#767] [Q&A]
I am writing to kindly request some information regarding the Prototyping Deck for the Crazyflie 2.0, specifically in relation to attaching an external servo motor and controlling it through firmware modifications.

My objective is to connect a 5V servo motor to the Crazyflie board, utilizing the capabilities of the Prototyping Deck. However, I have encountered difficulties in finding comprehensive instructions on how to properly attach or solder the servo motor to the Prototyping Deck. Therefore, I would greatly appreciate your assistance in clarifying the following points:

Attachment Method: Could you please advise whether attaching the servo motor directly through soldering to the Prototyping Deck would be sufficient, or if additional electronic components are required for proper functionality?

Pin Configuration: I kindly request guidance on which specific pins I should connect the 5V servo motor to on the Prototyping Deck. Understanding the correct pin configuration is crucial to ensure seamless integration and control.

Schematics: If possible, it would be immensely helpful if you could provide any available schematics or documentation outlining the recommended attachment process and pin connections for the servo motor on the Prototyping Deck. Moreover, it would greatly facilitate the process if you could indicate the specific pin connections between the middle section of the Prototyping deck and the corresponding pins on the crazyflie.

Your support in clarifying these queries would greatly aid me in successfully incorporating the servo motor into my project. I genuinely appreciate your time and assistance. Thank you very much for your attention to this matter. I look forward to your prompt response.

**Comment by ataffanel** [comment]
Hi,

While it should be possible, you will have a couple of problem to solve:
 - You want to power the servo from Vcom and not from VCC. Vcom is the battery voltage when the Crazyflie is not charging and the USB voltage when charging. This is the line with the most power available on the Crazyflie (VCC is limited to about 100mA).
 - Crazyflie runs on a single cell LiPO battery, this means that the battery voltage will be from 4.2 to 3.0V. This is likely not high enough to power a 5V servo so you likely need a Boost or a Buck-Boost converter in order to generate the 5V power supply for the servo
 - The PWM input line of the servo can be connected directly to a PWM-capable pin of the Crazyflie expansion connector. In the big-quad deck we are using IO_2, IO_3, RX2, TX2 as PWM capable pins
 - You will have to implement the PWM-driving code in the Crazyflie using the ST lib. Examples of that can be found in the Crazyflie (the brushless ESC driving code should be very similar). This can be implemented as a Crazyflie app.

The BigQuad deck actually implements 4 servo output, they are intended to be connected to ESCs but the interface is exactly the same as for regular servo, you can check the schematic for reference: https://www.bitcraze.io/documentation/hardware/big_quad_deck/bigquad-rev-c1.pdf. As you can see there is one line named `5V_IN`, this is going to be your main challenge, you need 5V to power the servo.

I hope this helps you going forward,
/Arnaud

**Reply by NarekHarutyu** [reply]
I would like to inquire about the following matter. For my project, I am planning to utilize the BigQuad deck to connect the crazyflie board to a larger quadcopter. Additionally, I intend to use the Prototyping deck to attach servo motors to the board. To power this setup, I will employ a substantial battery, which should provide sufficient voltage without requiring a Boost module. Please correct me if I have misunderstood any aspects of your previous message.

Firstly, am I correct in assuming that I can connect the power wire of the servo to the VCOM pin in order to obtain the necessary voltage for servo operation? Furthermore, should I connect the signal wire of the servo to one of the PWM pins on the crazyflie board, such as the STM32F4 GPIOs? If not, kindly advise me on which pins to utilize for this purpose.

Moreover, I find myself somewhat confused by the Prototyping deck schematics. Once I have placed the deck on top of the crazyflie board, how should I proceed with attaching and soldering the power, signal, and ground wires of the servo? Should I utilize the 40 pins located in the middle, or should I directly solder them to the corresponding holes on the Prototyping deck through which the actual crazyflie pins are coming? If the latter is not the correct approach, could you please provide a clearer explanation of how to properly attach the wires to the Prototyping deck? Specifically, I would appreciate a detailed explanation of how the middle holes function in conjunction with the pins. Are they similar to the way a breadboard operates, or do they possess unique properties?

**Reply by krichardsson** [reply]
The comment from @herryyy65 seems to be AI-generated

**Reply by ataffanel** [reply]
@NarekHarutyu If you are already using the big-quad deck, you should not have 5V issues as long as you already provide the Crazyflie with 5V already. As for the PWM pins, I would suggest looking in the STM32F405 datasheet together with the Crazyflie 2.1 schematic and identify what other pins are connected to a timer output-compare channel. This would allow you to identify if any other pin are available to control a servo-motor.

I can answer your prototyping board question on your new thread in #774.

By the way, the information that you are on a big-quad using the big-quad deck would have been very useful from the beginning. My original message was all based on the assumption that you where using the Crazyflie without any additional deck or hardware. This information would have saved us both quite some time.
